In [1]:
#index, year, month, intent, police, sex, age, race, hispanic, place, education

In [1]:
import csv 
f=open("guns.csv","r")
csvreader=csv.reader(f)
data=list(csvreader)
print(data[:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [2]:
headers=data[0]
print(headers)
data=data[1:]
print(data[0:5])

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2']]


In [3]:
years=[event[1] for event in data]
year_counts={}
for event in years:
    if event in year_counts:
        year_counts[event]=year_counts[event]+1
    else:
        year_counts[event]=1
print(year_counts)

{'2014': 33599, '2012': 33563, '2013': 33636}


In [4]:
import datetime

dates=[datetime.datetime(year=int(event[1]),month=int(event[2]), day=1 ) for event in data]
dates[0:5]

date_counts={}
for an_event in dates:
    if an_event in date_counts:
        date_counts[an_event]=date_counts[an_event]+1
    else:
        date_counts[an_event]=0
            
print(date_counts)

{datetime.datetime(2012, 2, 1, 0, 0): 2356, datetime.datetime(2012, 4, 1, 0, 0): 2794, datetime.datetime(2012, 3, 1, 0, 0): 2742, datetime.datetime(2012, 10, 1, 0, 0): 2732, datetime.datetime(2014, 10, 1, 0, 0): 2864, datetime.datetime(2013, 2, 1, 0, 0): 2374, datetime.datetime(2013, 6, 1, 0, 0): 2919, datetime.datetime(2013, 5, 1, 0, 0): 2805, datetime.datetime(2013, 7, 1, 0, 0): 3078, datetime.datetime(2012, 1, 1, 0, 0): 2757, datetime.datetime(2014, 4, 1, 0, 0): 2861, datetime.datetime(2012, 12, 1, 0, 0): 2790, datetime.datetime(2014, 8, 1, 0, 0): 2969, datetime.datetime(2013, 11, 1, 0, 0): 2757, datetime.datetime(2014, 12, 1, 0, 0): 2856, datetime.datetime(2013, 10, 1, 0, 0): 2807, datetime.datetime(2012, 5, 1, 0, 0): 2998, datetime.datetime(2014, 2, 1, 0, 0): 2360, datetime.datetime(2013, 3, 1, 0, 0): 2861, datetime.datetime(2012, 8, 1, 0, 0): 2953, datetime.datetime(2014, 1, 1, 0, 0): 2650, datetime.datetime(2014, 6, 1, 0, 0): 2930, datetime.datetime(2012, 6, 1, 0, 0): 2825, date

In [5]:
sex_counts={}
for event in data: 
    if event[5] in sex_counts:
        sex_counts[event[5]]=sex_counts[event[5]]+1
    else:
        sex_counts[event[5]]=0

race_counts={}
for event in data: 
    if event[7] in race_counts:
        race_counts[event[7]]=race_counts[event[7]]+1
    else:
        race_counts[event[7]]=0
        
print(sex_counts)
print(race_counts)

{'M': 86348, 'F': 14448}
{'Hispanic': 9021, 'Native American/Native Alaskan': 916, 'Black': 23295, 'Asian/Pacific Islander': 1325, 'White': 66236}


In [6]:
max_deaths=0
month_of_deaths=None
for ind,value in date_counts.items():
    if value > max_deaths:
        max_deaths= value
        month_of_deaths=ind

print("The most deaths took place on " + str(month_of_deaths)+ " of " + str(max_deaths) + " people")


The most deaths took place on 2013-07-01 00:00:00 of 3078 people


In [7]:
f2=open("census.csv","r")
csvreader2=csv.reader(f2)
census=list(csvreader2)

In [8]:
mapping={
    "Asian/Pacific Islander":int(census[1][13])+int(census[1][14]),
    "Black":int(census[1][12]),
    "Native American/Native Alaskan":int(census[1][13]),
    "Hispanic":int(census[1][11]),
    "White":int(census[1][10])
}
print(mapping)

{'Native American/Native Alaskan': 3739506, 'Hispanic': 44618105, 'White': 197318956, 'Asian/Pacific Islander': 18899022, 'Black': 40250635}


Based on the "gun.csv" file, it can be determined that a majority of gun deaths are of white males. Overall, gun deaths were pretty consistent throughout 2012-2014. The most deaths that took place within this timeframe was in July 2013 with a total of 3,078 gun deaths.

In [9]:
race_per_hundredk={}

for key,value in race_counts.items():
    race_per_hundredk[key]=(value*100000)/mapping[key]

print(race_per_hundredk)

{'Native American/Native Alaskan': 24.495214073730594, 'Black': 57.87486334066531, 'White': 33.567986240511026, 'Asian/Pacific Islander': 7.010944799154157, 'Hispanic': 20.2182499682584}


In [14]:
intents=[event[3] for event in data]
races=[event[7] for event in data]

homicide_race_counts={}
for i,race in enumerate(races):
    if intents[i]=="Homicide":
        if race in homicide_race_counts:
            homicide_race_counts[race]=homicide_race_counts[race]+1
        else: 
            homicide_race_counts[race]=0

print(homicide_race_counts)

for key,value in homicide_race_counts.items():
    homicide_race_counts[key]=(value*100000)/mapping[key]

print(homicide_race_counts)
  

{'Hispanic': 5633, 'Native American/Native Alaskan': 325, 'Black': 19509, 'Asian/Pacific Islander': 558, 'White': 9146}
{'Hispanic': 12.624919861567406, 'Native American/Native Alaskan': 8.690987526159873, 'Black': 48.468800554326656, 'Asian/Pacific Islander': 2.9525337342852978, 'White': 4.635135004464548}


In considering the gun deaths that occurred between 2012-2014, the race most affected by homicides by population percentage was the Black community. 

I would love to continue investigating that data and determine if education, place, police, and age affect the frequency of the gun deaths. 

Other steps: 
- figure out the link i